src/main/scala/net/cakesolutions/CakeBuildPlugin.scala

```diff
@@ -58,13 +65,23 @@ object CakeBuildPlugin extends AutoPlugin {
       sys.env.get("SBT_VOLATILE_TARGET") match {
         case None => target.value
         case Some(base) =>
+          val baseDir = new File(base)
+          assert(
```

Comment:
I don't think many of these checks are necessary as the dir is allowed to not exist in many cases.

Can you perhaps put the `requires` and raising of exceptions into a separate PR instead of mixing it up with the docs?